documentation/modules/oauth/con-oauth-server-config.adoc

```diff
@@ -0,0 +1,160 @@
+      type: custom
+      sasl: true
+      listenerConfig:
+        sasl.enabled.mechanisms: OAUTHBEARER
+        oauthbearer.sasl.server.callback.handler.class: io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler
+        oauthbearer.sasl.jaas.config: |
+          org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required
+            oauth.valid.issuer.uri="https://${SSO_HOST}/realms/kafka-authz"
+            oauth.jwks.endpoint.uri="https://${SSO_HOST}/realms/kafka-authz/protocol/openid-connect/certs"
+            oauth.username.claim="preferred_username"
+            oauth.ssl.truststore.location="/mnt/oauth-certs/tls.crt"
+            oauth.ssl.truststore.type="PEM";
+        connections.max.reauth.ms: 3600
+----
+
+In this configuration:
+
+* The listener uses the `OAUTHBEARER` SASL mechanism.
+* Tokens are validated locally using the JWKS endpoint published by the authorization server.
+* The JAAS module and callback handler perform token validation and map a specified token claim to the Kafka principal.
+* The truststore is mounted into the Kafka pod and used for TLS connections to the JWKS endpoint.
+* The listener requires periodic re-authentication.
+The connections.max.reauth.ms setting ensures that the broker prompts clients to re-authenticate at regular intervals, allowing expiring OAuth access tokens to be refreshed without closing the connection.
+
+=== Token introspection example
+
+The following example shows a listener configuration that validates opaque access tokens using an OAuth 2.0 introspection endpoint. 
+The Kafka broker authenticates to the authorization server using a client ID and a client secret mounted from a Kubernetes secret.
+
+[source,yaml]
+----
+# ...
+spec:
+  kafka:
+    listeners:
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+        authentication:
+          type: custom
+          sasl: true
+          listenerConfig:
+            sasl.enabled.mechanisms: OAUTHBEARER
+            oauthbearer.sasl.server.callback.handler.class: io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler
+            oauthbearer.sasl.jaas.config: |
+              org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required
+                oauth.access.token.is.jwt="false"
+                oauth.introspection.endpoint.uri="https://${SSO_HOST}/realms/myrealm/protocol/openid-connect/token/introspect"
+                oauth.client.id="kafka-broker"
+                oauth.client.secret="/opt/oauth/credentials/client-secret"
+                oauth.valid.issuer.uri="https://${SSO_HOST}/realms/myrealm"
+                oauth.username.claim="preferred_username"
+                oauth.ssl.truststore.location="/opt/oauth-certs/tls.crt"
+                oauth.ssl.truststore.type="PEM";
+    template:
+      pod:
+        volumes:
+          - name: oauth-credentials
+            secret:
+              name: kafka-broker-oauth-secret
```

Comment:
I guess this volume mounts the `/opt/oauth/credentials/client-secret` file. The way to use that approach would be to map the content of this file to the value of the environment variable `OAUTH_CLIENT_SECRET`. Something like:
```
env:
  - name: OAUTH_CLIENT_SECRET
    valueFrom:
      secretKeyRef:
        name: kafka-broker-auth-secret
        key: secret
```

Reply:
I've removed the volume config and added a note on loading configuration values from external sources